components/data-table/index.jsx

```diff
@@ -131,23 +131,29 @@ const DataTable = createReactClass({
 		 */
 		fixedLayout: PropTypes.bool,
 		/**
-		 * The collection of items to render in the table.
+		 * The collection of items to render in the table. This is an array of objects with each object having keys that correspond with the  `property` prop of each `DataTableColumn`.
 		 */
-		items: PropTypes.array.isRequired,
+		items: PropTypes.arrayOf(
+			PropTypes.shape({
+				id: PropTypes.string.isRequired,
+			})
+		).isRequired,
 		/**
 		 * A variant which removes hover style on rows
 		 */
 		noRowHover: PropTypes.bool,
 		/**
-		 * This function fires when the selection of rows changes.
+		 * This function fires when the selection of rows changes. This component passes in `event, { selection }` to the function. `selection` is an array of objects from the `items` prop.
+		 *
+		 * This used to be `onChange` which is deprecated now, so that the parameters can be consistent with other components. `onChange` passed in the selection first and the event wtihout a data object.
 		 */
-		onChange: PropTypes.func,
```

Comment:
Is this _removing_ the onChange prop?  Shouldn't we leave it in if its only deprecated, and then remove it later?